### PR TITLE
Update the number of nodes in the EKS cluster

### PR DESCRIPTION
**Pull request by ribeiro-rodrigo**
This functionality allows changing the number of nodes in the EKS cluster. 

Types of changes
New feature (non-breaking change which adds functionality)

Linked Issues
rancher/rancher#27847

### CODE REVIEW

File: drivers/eks/eks_driver.go [hunk 1]
```diff
@@ -1289,6 +1315,87 @@ func (d *Driver) SetVersion(ctx context.Context, info *types.ClusterInfo, versio
...
+		),
+	})
+	if err != nil {
+		return err
+	}
+
+	stackName := getWorkNodeName(state.DisplayName)
+	cloudFormationService := cloudformation.New(sess)
+
+	templateOutput, errGetTemplate := cloudFormationService.GetTemplate(&cloudformation.GetTemplateInput{
```
Comment (rmweir) [inline drivers/eks/eks_driver.go:1352]:
```suggestion
	templateOutput, err := cloudFormationService.GetTemplate(&cloudformation.GetTemplateInput{
```

File: drivers/eks/eks_driver.go [hunk 2]
```diff
@@ -1289,6 +1315,87 @@ func (d *Driver) SetVersion(ctx context.Context, info *types.ClusterInfo, versio
...
+		StackName: aws.String(stackName),
+	})
+
```
Comment (rmweir) [inline drivers/eks/eks_driver.go:1355]: remove new line

File: drivers/eks/eks_driver.go [hunk 3]
```diff
@@ -1289,6 +1315,87 @@ func (d *Driver) SetVersion(ctx context.Context, info *types.ClusterInfo, versio
...
+	}
+
+	updateInput := &cloudformation.UpdateStackInput{
+		Capabilities: []*string{aws.String(cloudformation.CapabilityCapabilityIam)},
+		StackName:    aws.String(stackName),
+		TemplateBody: templateOutput.TemplateBody,
+		Parameters:   params,
+	}
+
+	_, errUpdate := cloudFormationService.UpdateStack(updateInput)
```
Comment (rmweir) [inline drivers/eks/eks_driver.go:1367]:
```suggestion
	_, err := cloudFormationService.UpdateStack(updateInput)
```

File: drivers/eks/eks_driver.go [hunk 4]
```diff
@@ -1289,6 +1315,87 @@ func (d *Driver) SetVersion(ctx context.Context, info *types.ClusterInfo, versio
...
+
```
Comment (rmweir) [inline drivers/eks/eks_driver.go:1368]: remove newline

File: drivers/eks/eks_driver.go [hunk 5]
```diff
@@ -989,6 +1007,14 @@ func (d *Driver) Update(ctx context.Context, info *types.ClusterInfo, options *t
 		state.ClientSecret = newState.ClientSecret
 	}
 
+	if newState.DesiredASGSize > 0 && newState.DesiredASGSize != state.DesiredASGSize {
```
Comment (rmweir) [inline drivers/eks/eks_driver.go:1010]: Can you move the newState.DesiredASGSize to the validate() function and have validate work for updates and create or create a separate function?

File: drivers/eks/eks_driver.go [hunk 6]
```diff
@@ -1289,6 +1315,87 @@ func (d *Driver) SetVersion(ctx context.Context, info *types.ClusterInfo, versio
 	return nil
 }
 
+func (d *Driver) updateNodePoolAndAwait(state state, info *types.ClusterInfo) error {
```
Comment (rmweir) [inline drivers/eks/eks_driver.go:1318]: I think the receiver can be removed
```suggestion
func updateNodePoolAndAwait(state state, info *types.ClusterInfo) error {
```

**Review by rmweir — CHANGES_REQUESTED** [review]
Can you change errors to be named err for consistency?

**Comment by deniseschannon** [comment]
As we are moving to the new EKS v2 support in Rancher v2.5, we encourage all users to upgrade and use the new provisioner which allows being able to change the number of nodes in EKS.

**Pull request closed without merging**